### DISCUSSION TITLE
Set color of table row conditionally

### DISCUSSION THREAD

**Discussion by emincer** [filamentphp/filament#7309] [Help]
Is there a way to check one of records of a table column and based on that, set the text color of the row? Like if I have a "status' column and the record is 'canceled', make the entire row's text red?

**Comment by RX-77IT** [comment]
With my basic knowledge, I'm afraid you will have to check the condition for the red text within each TextColumn.
You could write a custom static functions in the ModelResource to avoid repeating too much code. Maybe something like...
```php
use Filament\Tables;
use Illuminate\Database\Eloquent\Model;
//...

class ModelResource extends Resource
{
    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('name')
                    ->extraAttributes(fn (?Model $record) => self::redTextColor($record)),
                Tables\Columns\TextColumn::make('surname')
                    ->extraAttributes(fn (?Model $record) => self::redTextColor($record)),
                Tables\Columns\TextColumn::make('role')
                    ->extraAttributes(fn (?Model $record) => self::redTextColor($record)),
            ])
        //...
    }

    private static function redTextColor(?Model $record): array
    {
        return $record->name == 'name' // example condition
            ? ['style' => 'color: #ff1111;']
            : [];
    }

    //...
}
```
If the condition is complex you could use an accessor for the model and cache it: https://laravel.com/docs/10.x/eloquent-mutators#defining-an-accessor


EDIT: nevermind, there is another cleaner method in the docs that I didn't know about. XD See the answer below mine.

**Comment by danharrin** [comment]
https://filamentphp.com/docs/2.x/tables/getting-started#record-classes

**Comment by emincer** [comment]
I swear I looked through the docs before asking. Don’t know why I couldn’t find this. Thanks Dan.Sent from my iPhoneOn Jul 31, 2023, at 3:01 AM, Dan Harrin ***@***.***> wrote:﻿
https://filamentphp.com/docs/2.x/tables/getting-started#record-classes
